File: src/python/algorithms/matrix/nQueens/n_queens.py
def solve_n_queens(n):
    def backtrack(r):
        if r == n:
            board = [''.join(row) for row in curr]
            result.append(board)
            return
        for c in range(n):
            if c in cols or (r+c) in pos_diag or (r-c) in neg_diag:
                continue
            cols.add(c)
            pos_diag.add(r+c)
            neg_diag.add(r-c)
            curr[r][c] = 'Q'
            backtrack(r+1)
            cols.remove(c)
            pos_diag.remove(r+c)
            neg_diag.remove(r-c)
            curr[r][c] = '.'
    result = []
    cols, pos_diag, neg_diag = set(), set(), set()
    curr = [['.']*n for _ in range(n)]
    backtrack(0)
    return result

if __name__ == "__main__":
    print("Input: n = 1\nOutput:", solve_n_queens(1))
    print("Input: n = 2\nOutput:", solve_n_queens(2))
    print("Input: n = 3\nOutput:", solve_n_queens(3))
    print("Input: n = 4\nOutput:", solve_n_queens(4))
    print("Input: n = 5\nOutput:", solve_n_queens(5))

File: src/javascript/algorithms/matrix/nQueens/nQueens.js
function solveNQueens(n) {
    const result = [];
    const cols = new Set(), posDiag = new Set(), negDiag = new Set();
    const board = Array.from({ length: n }, () => Array(n).fill('.'));
    function backtrack(r) {
        if (r === n) {
            result.push(board.map(row => row.join('')));
            return;
        }
        for (let c = 0; c < n; c++) {
            if (cols.has(c) || posDiag.has(r + c) || negDiag.has(r - c)) continue;
            cols.add(c);
            posDiag.add(r + c);
            negDiag.add(r - c);
            board[r][c] = 'Q';
            backtrack(r + 1);
            cols.delete(c);
            posDiag.delete(r + c);
            negDiag.delete(r - c);
            board[r][c] = '.';
        }
    }
    backtrack(0);
    return result;
}

console.log("Input: n = 1\nOutput:", solveNQueens(1));
console.log("Input: n = 2\nOutput:", solveNQueens(2));
console.log("Input: n = 3\nOutput:", solveNQueens(3));
console.log("Input: n = 4\nOutput:", solveNQueens(4));
console.log("Input: n = 5\nOutput:", solveNQueens(5));
